### DISCUSSION TITLE
HowTo: getReport from SP-API using MWS reportRequestId

### DISCUSSION THREAD

**Discussion by JordanForeman** [amzn/selling-partner-api-models#3148] [General] [answered]
:wave: hello! I'm in the process of migrating existing customers to the new Selling Partner API. These customer's integrations are currently powered by MWS, and part of the integration is the implementation of the asynchronous `request - check - fetch` Report workflow (documented below):

![image](https://user-images.githubusercontent.com/1462087/135490131-da92f53f-2e23-41fe-99df-384202e4c560.png)

I'm currently caching the `reportRequestId` and using it to load the report when it's ready. I'm wondering if anyone has a recommendation for using this `reportRequestId` to get the report from SP? The process is still asynchronous, but the concept of a "report request" no longer exists - instead the `createReport` API returns the ID of the report to be created, and `getReport` is used to poll for status changes. Can I use the `reportRequestId` to get this same entity?

I'll be migrating our report generation process to also using the SP API, but I'm hoping to account for customers who will have reports requested _before_ cutover and fetching their reports _after_ cutover.

**Comment by ezpinsky** [comment, marked as answer]
Please try to read the documentation [here](https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/use-case-guides/reports-api-use-case-guide/reports-api-use-case-guide_2021-06-30.md#step-3-retrieve-the-report) to help you migrate. In short, you can not just use the reportRequestId. In SP-API, you use createReport operation to get a reportId, you then  poll a report using the getReport operation passing the reportId and if you receive a DONE status, you will also receive a reportDocumentId in the response. You will need to cache the reportDocumentId and use that with the getReportDocument operation to receive a url where you're report can be downloaded from. Hope this helps.

**Reply by JordanForeman** [reply]
Thanks Ezra, but that doesn't really help. I know how to use the _new_ API. What I don't know is if I can leverage the existsing `reportRequest` entity (and its ID, which I currently have) to ease the migration to the new API

> In short, you can not just use the reportRequestId

That is my assumption as well, but would like to be certain.

**Reply by JordanForeman** [reply]
I suppose this answer confirms my suspicions and in lieu of any alternatives I will consider the answer to be:

> you can not just use the `reportRequestId`

Thanks Ezra
